### ISSUE TITLE
[New Plugin]: binaryninja-ollama

### ISSUE THREAD

**Issue by ahaggard2013** [Vector35/community-plugins#274]
### Plugin Description

Plugin to rename functions and variables with a locally hosted ollama server. Can use any of the openly available models such as llama3 and gemma2.

### Plugin Repository URL

Repo URL: https://github.com/ahaggard2013/binaryninja-ollama

### Real Release

- [X] I have made a real release

**Comment by ahaggard2013** [comment]
read me was overwritten by release script, reposting

**Comment by psifertex** [comment]
No worries, I have some other feedback i'm working on.

**Comment by psifertex** [comment]
Here's my comments so far:

Thanks for the submission.

Should https://github.com/ahaggard2013/binaryninja-ollama/blob/main/ui.py#L28C43-L28C57 be updated to use 127.0.0.1 instead?

Also, please [remove PySide6](https://github.com/ahaggard2013/binaryninja-ollama/blob/main/requirements.txt#L1) in the requirements.txt as that is automatically provided by Binary Ninja if you first `import binaryninjaui` and then `import PySide6`. Otherwise you'll be getting whatever version of PySide6 Python installs which is likely to cause issues.

Alternatively, you could replace the UI code with [interaction API calls](https://github.com/Vector35/binaryninja-api/blob/dev/python/examples/export_svg.py#L46-L72) so you don't have to write pyside at all.

Not important suggestion:
the documentation for functions is nice, but if you're going to do that you should really consider [type hinting](https://docs.python.org/3/library/typing.html) as it will help linters and IDEs make sure you code has less bugs in the future!

**Comment by ahaggard2013** [comment]
thanks, fixed the ip/requirements, porting to interaction calls and type hinting will take a a bit more effort, but will look into it!

**Comment by negasora** [comment]
Reopening this - please make a release with the requirements fixes

**Comment by psifertex** [comment]
TL;DR your latest code has the change to requirements, but not the latest release actually cut. This actually caused a user who installed the plugin to have all UI plugins break unfortunately so we've removed it from the plugin manager until it's resolved.

**Comment by ahaggard2013** [comment]
whoops! I'm new to configuring releases, should be fixed now.

**Comment by psifertex** [comment]
Thanks -- added back. Also, have you tried https://github.com/mrphrazer/reverser_ai ? I don't think I mentioned it before but it also uses an ollama model. Might be worth consolidating/merging any shared features, though Tim's plugin is GPLv2 so that could be an issue.

**Comment by ahaggard2013** [comment]
I didn't see that plugin! That's awesome, very similar. I originally only saw the openai one which sparked the effort. I will have to give it a try later, looks like it has some cool additional features.
